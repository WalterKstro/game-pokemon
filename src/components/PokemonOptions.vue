<template>
  <aside>
      <h2 class="list-title">Seleccione una opci√≥n</h2>
      <ul class="container-list">
        <li class="list-item" 
            v-for="({id,name}) in arrayOptionsPokemons" 
            :key="id">
            <button class="list-button" @click="selectPokemon($event,id)" :data-id="id">{{name}}</button>
        </li>
        </ul>
  </aside>
</template>

<script>
export default {
    name: 'PokemonOptions',
    props: {
        arrayOptionsPokemons: {
            type: Array,
            required: true,
        }
    },
    methods: {
        selectPokemon($event,idPokemon){
            const arrayButtonOptions = document.querySelectorAll('[data-id]');
            const idButtonClicked = $event.target.dataset.id;
            arrayButtonOptions.forEach(button => {
                (button.dataset.id == idButtonClicked) ? button.classList.add('selected') : button.setAttribute('disabled',true);
            });
            this.$emit('select-pokemon', idPokemon);
        }
    }
}
</script>

<style>
.container-list{
   padding: 0;
   text-indent: 0;
   list-style-type: none;
    max-width: 200px;
    min-width: 100px;
    width: 100%;
    margin-right: auto;
    margin-left: auto;
}
.list-item{
    margin: 5px 0;
}
.list-button{
    border-radius: 3px;
    width: 100%;
    padding: 10px 0;
    border: 0;
    cursor: pointer;
    background-color: #263238;
    color: #feca1b;
}
.list-button:hover{
    background-color: #feca1b;
    color: #263238;
}
.list-title{text-align: center;}
.selected{
    background-color: #feca1b;
    color: #263238;
}
</style>