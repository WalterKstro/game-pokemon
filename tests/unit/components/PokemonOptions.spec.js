import PokemonOptions from '@/components/PokemonOptions';
import { shallowMount } from '@vue/test-utils';

describe('Test suit on component <PokemonOptions/>', () => {
    let wrapper;
    const arrayPokemons = [
        { id: 196, name: 'espeon' },
        { id: 630, name: 'mandibuzz' },
        { id: 173, name: 'cleffa' },
        { id: 238, name: 'smoochum' }
    ];

    beforeEach(() => {
        wrapper = shallowMount(PokemonOptions,{
            propsData: {
                arrayOptionsPokemons: arrayPokemons,
            }
        });
    });

    test('Should create a correct snapshot of <PokemonOptions/>',() => {
        expect(wrapper.html()).toMatchSnapshot();
    })

    test('Should display four elements <button>', () => {
        const arrayButtons = wrapper.findAll('.list-button');
        const firstElement = arrayButtons[0];
        const lastElement = arrayButtons[3];

        expect(arrayButtons.length).toBe(4);
        expect(firstElement.text()).toBe('espeon');
        expect(lastElement.text()).toBe('smoochum');
        
    });

    test('Should emit an event(select-pokemon) when click on a pokemon', () => {
        const [button1, button2, button3, button4] = wrapper.findAll('.list-button');

        button1.trigger('click');
        button2.trigger('click');
        button3.trigger('click');
        button4.trigger('click');

        const [ param1, param2, param3, param4 ] = wrapper.emitted('select-pokemon');
        

        expect(param1).toEqual([arrayPokemons[0].id]);
        expect(param2).toEqual([arrayPokemons[1].id]);
        expect(param3).toEqual([arrayPokemons[2].id]);
        expect(param4).toEqual([arrayPokemons[3].id]);

    })
})
