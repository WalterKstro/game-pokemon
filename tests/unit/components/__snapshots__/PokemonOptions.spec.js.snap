// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suit on component <PokemonOptions/> Should create a correct snapshot of <PokemonOptions/> 1`] = `
<aside>
  <h2 class="list-title">Seleccione una opci√≥n</h2>
  <ul class="container-list">
    <li class="list-item"><button class="list-button" data-id="196">espeon</button></li>
    <li class="list-item"><button class="list-button" data-id="630">mandibuzz</button></li>
    <li class="list-item"><button class="list-button" data-id="173">cleffa</button></li>
    <li class="list-item"><button class="list-button" data-id="238">smoochum</button></li>
  </ul>
</aside>
`;
